lib/RenderApp/Controller/IO.pm

```diff
@@ -144,6 +145,11 @@ sub catalog {
         return ( -e $root_path ) ? $c->rendered(200) : $c->rendered(404);
     }
 
+    if ( !($root_path =~ /^(:?webwork-open-problem-library\/|private\/)/) ) {
```

Comment:
Since we have this I will have to add extra validation:
Example: CUNY/path/to/pg.pg
first I try the path itself `CUNY/path/to/pg.pg`
Then I try `Contrib/CUNY/path/to/pg.pg`

But my first one will throw an error now instead because instead of getting a response it will get an error

I'm ok with that I just don't know if this check is all that helpful and if we expand in the future we will need to update this

Also you forgot Library so calls that use Library will fail

Reply:
Tested and this comment is void, everything worked as expected